cluster/gce/config-test.sh

```diff
@@ -505,7 +505,7 @@ ROTATE_CERTIFICATES="${ROTATE_CERTIFICATES:-}"
 # into kube-controller-manager via `--concurrent-service-syncs`
 CONCURRENT_SERVICE_SYNCS="${CONCURRENT_SERVICE_SYNCS:-}"
 
-SERVICEACCOUNT_ISSUER="https://kubernetes.io/${CLUSTER_NAME}"
+SERVICEACCOUNT_ISSUER="https://kubernetes.default.svc"
```

Comment:
Does "we can't change it in `config-default.sh`" mean "we can't change it in `config-test.sh`"?

Is there a way to detect the current value and only use this as a default (e.g. for new clusters)?

(In general: is there no way to change the default of a flag if changing it is backwards-incompatible?)

Reply:
> Does "we can't change it in config-default.sh" mean "we can't change it in config-test.sh"?

Not strictly. 

> Is there a way to detect the current value and only use this as a default (e.g. for new clusters)?

I don't think so. We have to assume we're on a fresh VM. Unless `SERVICEACCOUNT_ISSUER` is being overridden by the user or cloud provider (pretty likely IMO, but can't guarantee), the current `kubernetes.io` value will be used. (There might be some nasty things we could do like read etcd directly (or the storage) to pull an iss claim off a default service account token...I wouldn't really be comfortable sneaking around the API server though).

> (In general: is there no way to change the default of a flag if changing it is backwards-incompatible?)

Yeah, generally true.

Reply:
Yeah... that said I think this should be ok just for the purpose of an e2e test? 
Where:
- We don't know what the external hostname is going to be, because it's just an ephemeral test.
- The test pod is using the discovery doc, which will report the hostname we set the issuer to, so we don't need to worry about the test failing because it uses the wrong hostname.

Reply:
If we leave this as-is, we should add a comment that this is not something that should be copied for real use, and that it is really only usable from inside pods

Reply:
Done.

Reply:
Moved to #88048